File: SupplierPortal.Web/Modules/ServerTypes/Items/ItemColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ItemRow } from "./ItemRow";

export interface ItemColumns {
    Id: Column<ItemRow>;
    ItemId: Column<ItemRow>;
    ItemName: Column<ItemRow>;
}

export class ItemColumns extends ColumnsBase<ItemRow> {
    static readonly columnsKey = 'Items.Item';
    static readonly Fields = fieldsProxy<ItemColumns>();
}

File: SupplierPortal.Web/Modules/Items/Item/ItemColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace SupplierPortal.Items.Columns;

[ColumnsScript("Items.Item")]
[BasedOnRow(typeof(ItemRow), CheckNames = true)]
public class ItemColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public int ItemId { get; set; }
    [EditLink]
    public string ItemName { get; set; }
}